ui/src/components/Nodes/ColumnSelectDropdown.vue

```diff
@@ -0,0 +1,78 @@
+<template>
+  <FeatherDropdown>
+    <template v-slot:trigger="{ attrs, on }">
+      <FeatherButton
+        icon="Select Columns"
+        v-bind="attrs"
+        v-on="on"
+      >
+        <FeatherIcon :icon="settingsIcon" class="node-actions-icon" />
+      </FeatherButton>
+    </template>
+    <div class="node-actions-reset">
+      <FeatherButton secondary @click="resetToDefault">Default</FeatherButton>
+    </div>
+    <FeatherDropdownItem
+      v-for="col in columns"
+      :key="col.id"
+      >
+        <div class="column-select-item-wrapper">
+          <FeatherCheckbox
+            class="checkbox"
+            @update:modelValue="selectColumn(col)"
+            :modelValue="col.selected"
+          >{{ col.label }}</FeatherCheckbox>
+        </div>
+    </FeatherDropdownItem>
+  </FeatherDropdown>
+</template>
+
+<script setup lang="ts">
+import { FeatherButton } from '@featherds/button'
+import { FeatherCheckbox } from '@featherds/checkbox'
+import { FeatherDropdown, FeatherDropdownItem } from '@featherds/dropdown'
+import { FeatherIcon } from '@featherds/icon'
+import Settings from '@featherds/icon/action/Settings'
+import { markRaw } from 'vue'
+import { useStore } from 'vuex'
+import { NodeColumnSelectionItem } from '@/types'
+
+const store = useStore()
+const settingsIcon = markRaw(Settings)
+
+const columns = computed<NodeColumnSelectionItem[]>(() => store.state.nodeStructureModule.columns)
+
+const selectColumn = (col: NodeColumnSelectionItem) => {
+  const newItem = {
+    ...col,
+    selected: !col.selected
+  }
+
+  store.dispatch('nodeStructureModule/updateNodeColumnSelection', newItem)
+  return false
+}
+
+const resetToDefault = () => {
+  store.dispatch('nodeStructureModule/resetColumnSelectionToDefault')
+  return false
```

Comment:
I might be missing something here, but why the return false for this & above? Some kind of type issue?

Reply:
To prevent event propagation, but I may have added some of these extraneously and will remove them or add a comment.

Reply:
Removed here and elsewhere in upcoming PR.